files/default/neutron-ha-tool.py

```diff
@@ -646,6 +662,29 @@ def wait_router_ports_active(qclient, router_id, target_host, maxtries=60):
                            (router_id, ", ".join(remaining)))
 
 
+def ssh_delete_router_namespace(target_host, router_id):
+    namespace = "qrouter-" + router_id
+
+    ssh_client = paramiko.SSHClient()
+    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh_client.load_system_host_keys()
+    ssh_client.connect(target_host)
+    stdin, stdout, stderr = ssh_client.exec_command("ip netns delete " +
+                                                    namespace)
```

Comment:
Well. As outlined in the commit message this is meant for the case where the l3-agent is not running on the node for some reason (in our case e.g. during the upgrade). So neutron won't be able to talk to it to delete the namespace.

When the l3-agent is running the ssh command is basically a noop, because the namespace is already deleted by the l3-agent as part of the "remove_router_from_l3_agent" request.